__tests__/json_decode_test.ml

```diff
@@ -344,6 +267,24 @@ describe "field" (fun () ->
     expectFn
       (field "b" string) (Js.Json.parseExn {| { "a": null, "b": null } |})
       |> toThrow);
+
+  Test.throws (field "foo" int) [Bool; Float; Int; String; Null; Array; Object];
+);
+
+describe "at" (fun () ->
+  let open Json in
+  let open! Decode in
+
+  test "at boolean" (fun () ->
+    expect @@
+      at ["a"; "x"; "y"] boolean (Js.Json.parseExn {| { "a": { "x" : { "y" : false } }, "b": false } |})
+      |> toEqual Js.false_);
+  test "field nullAs" (fun () ->
+    expect @@
+      at ["a"; "x"] (nullAs Js.null) (Js.Json.parseExn {| { "a": { "x" : null }, "b": null } |})
```

Comment:
Could you put the json input for these tests into a separate variable? It gets a bit muddy with all the nested curlies and such.